### ISSUE TITLE
Allow user to select to force the reload of a stage when media is assigned/unassigned to it; new media edit page (HTML backend)

### ISSUE THREAD

**Issue by foobarlab** [foobarlab/UpStage-Video-Hack#97]
Whenever new media is assigned/unassigned to a stage the user should have the choice to forcibly let the stage reload for all connected clients.

**Comment by helenvarley** [comment]
yes - this is how the buttons on the stage edit page "should" work: there is one button that says "save stage", which should save the changes but not force reload; and another button that says "save and reload stage".

**Comment by foobarlab** [comment]
just to not mix up things: this ticket is just about the new media edit page for the dialog "assign media to stage". if there is an issue with the stage edit page there should be opened a new ticket...

**Comment by foobarlab** [comment]
created new ticket #99 for save stage bug

**Comment by foobarlab** [comment]
fixed reloading stages after assign/unassign (8c55047e7135e4e51a802535c4ec76dc03fc189b).
because it does not make sense to not reload the stages (and thus leaving the choice to the user) the changed stages are reloaded automatically.

**Comment by helenvarley** [comment]
actually it should be a choice; because right now when several of us are working on the same stage, & assigning & unassigning media, it becomes very irritating that the stage keeps reloading. it should be possible to do all your assigning/unassigning in the workshop and then have a single reload.
there is a checkbox option when you edit the name/voice etc for an avatar, it would be good to have this checkbox on the assigning part as well.

**Comment by foobarlab** [comment]
Making stage reloading optional has the consequence that people have to reload a stage manually. Doing it automatically prevents an out-of-sync state of a stage. Anyway making a reload optional is not a problem.

**Comment by helenvarley** [comment]
if it's optional then you can choose to force everyone to reload (i.e. if for some reason you need to do this during a performance); or you can choose not to force reload, for example when we are rehearsing & people are both on the stage & in the workshop. then it's better to do all the assigning/unassigning, & at a certain point before continuing with the rehearsal, warn everyone & reload the stage just one time. it doesn't matter in that situation if it has been out of sync while we are rehearsing/testing, it's more annoying if you are practising something on the stage & it keeps reloading without warning.

**Comment by foobarlab** [comment]
In general I think it is more user-friendly if no option is given. The one assigning or unassigning media could tell others that the stage will get reloaded. There may be more troubles when a stage gets out-of-sync and players or the audience report something not working. And then telling everyone to manually reload a stage is probably producing more chaos than forcing an automatic reload. Just my two cents.

**Comment by foobarlab** [comment]
Also it could be handy (if this gets optional on the assign/unassign panel of the edit media page) to have a button to manually force reload of a stage. Right now there seems to be a button for each stage to "save&reload" (appearently also working). So forcing an automatic reload whenever media gets assigned or unassigned seems not necessary and therefore making it optional is not that bad. But the consequence of having out-of-sync stage states remains and still could cause troubles when it is forgotten to get stages reloaded manually (eiuther by telling people or clicking the save&reload button on the stage edit page).

**Comment by helenvarley** [comment]
but the thing is that it forces a reload for each media item; so if you need to, for example, assign 5 media items and take 2 off, then it will reload 7 times. it would be better if you could select the items to assign & unassign, then click one button to action that & reload the stage (which is in fact how the old system with the check-boxes worked). maybe this should be resolved to the batch-processing ticket?

**Comment by foobarlab** [comment]
Open a new ticket then or watch out for existing tickets for the new feature of "multiple assignments/unassignments on the media edit page" (or similiar). Thanks!

**Comment by foobarlab** [comment]
Anyway I will make it optional to forcibly reload a stage when media is assigned or unassigned.

**Comment by foobarlab** [comment]
Related to #102.
